platformsh/demo-project
# 363 enable light/dark mode with a preference for user system settings

# How to test locally

## Start the demo app locally
1. **(optional)** Start the python backend server using `npm run start:backend`
2. Start the frontend: `REACT_APP_ENABLE_DESIGN_DEBUG=true npm run start:frontend`
   * The `ENABLE_DESIGN_DEBUG` variable enables you to emulate each step of the demo app

## Test this MR
This MR adds light/dark mode functionality to the demo app to support developers who may have a preference or struggle to read in one mode or the others. By default, the mode will default to the system preference provided by the operating system of the user.

Once the user overrides with their preference, it is stored in a local storage key named `theme`.

### Testing system mode
1. Open the demo app
2. It should match your operating system preference.
3. Change your operating system preference, the app should update to match (sometimes a page refresh is needed).

### Testing preferred setting
1. Select "light" or "dark" mode.
2. Refresh the page
3. The selection should persist despite your OS preference.
